stdlib/ref.mc

```diff
@@ -0,0 +1,8 @@
+-- Construct a reference
+let ref = lam x. tensorCreate [] (lam _. x)
+
+-- De-reference
+let deref = lam t. tensorGetExn t []
+
+-- Mutate reference
+let modref = lam t. lam v. tensorSetExn t [] v
```

Comment:
If these are added, shouldn't we remove the existing intrinsics? Otherwise the existing intrinsics are used unless this file is included, as can be observed in the REPL (because the intrinsic references are printed differently from a rank 0 tensor):
```
>> ref 5
(ref)
>> include "ref.mc"
>> ref 5
tensor
```

Reply:
I didn't want to remove stuff from the language before this PR was accepted. But the idea is to do what you are saying if no one objects.